plugins/node/opentelemetry-instrumentation-generic-pool/package.json

```diff
@@ -63,7 +64,6 @@
   "dependencies": {
     "@opentelemetry/instrumentation": "^0.40.0",
     "@opentelemetry/semantic-conventions": "^1.0.0",
-    "@types/generic-pool": "^3.1.9",
```

Comment:
Currently this package actually has to be a regular dependency as a type from this package is exported publicly from the package `index.d.ts`.

For end users installing this instrumentation in there application, typescript will attempt to import types from `index.d.ts`, as specified in [`package.json`](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/ea865837f88121973ac30d9420f21be262b0a669/plugins/node/opentelemetry-instrumentation-generic-pool/package.json#L6) ):

```
  "types": "build/src/index.d.ts",
```

It will find this content in `index.d.ts`:
```ts
import GenericPoolInstrumentation from './instrumentation';
export { GenericPoolInstrumentation };
export default GenericPoolInstrumentation;
//# sourceMappingURL=index.d.ts.map
```

And will follow `import GenericPoolInstrumentation from './instrumentation';` and find the following content in `instrumentation.d.ts`:
```ts
import { InstrumentationBase, InstrumentationConfig, InstrumentationNodeModuleDefinition } from '@opentelemetry/instrumentation';
import type * as genericPool from 'generic-pool';
export default class Instrumentation extends InstrumentationBase<typeof genericPool> {
    private _isDisabled;
    constructor(config?: InstrumentationConfig);
    init(): (InstrumentationNodeModuleDefinition<typeof genericPool> | InstrumentationNodeModuleDefinition<any>)[];
    private _acquirePatcher;
    private _poolWrapper;
    private _acquireWithCallbacksPatcher;
}
//# sourceMappingURL=instrumentation.d.ts.map
```

Once it hit the line 
```ts
import type * as genericPool from 'generic-pool';
```

Typescript will emit an error as the types are missing.

You can follow [this documentation](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/GUIDELINES.md#dependencies) to fix the issue. Basically what you need to do is to remove `import type * as genericPool from 'generic-pool';` from `instrumentation.ts` and replace `typeof genericPool` in code with `any`.